gammapy/background/energy_offset_array.py

```diff
@@ -71,7 +95,11 @@ def _fill_one_event_list(self, events):
         ev_cube_hist, ev_cube_edges = np.histogramdd(ev_cube_array,
                                                      [self.energy.value,
                                                       self.offset.value])
-        return ev_cube_hist
+        energy_bin = np.diff(ev_cube_edges[0])
+        solid_angle_bin = np.pi * np.diff(ev_cube_edges[1] ** 2)
+        solid_angle_tab, energy_tab = np.meshgrid(solid_angle_bin, energy_bin)
+        bkg_rate = ev_cube_hist / (solid_angle_tab * energy_tab)
```

Comment:
Let's postpone discussing if the formula is correct a bit.

For now, please put this into a separate method, so that filling events and computing the rate become calculations that can be run independently.

Probably it would even be best if a separate class is introduced that holds three arrays:
1. counts
2. livelime
3. rate
like we did here:
https://gammapy.readthedocs.org/en/latest/api/gammapy.background.CubeBackgroundModel.html

To compute background rates it's necessary to stack counts and livetime from runs, and only at the end one does the rate for the stacked cubes, right?